p2p/debug/connections.go

```diff
@@ -0,0 +1,149 @@
+package debug
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	libp2pNetwork "github.com/libp2p/go-libp2p/core/network"
+	libp2pPeer "github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/pokt-network/pocket/p2p/providers/peerstore_provider"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/p2p/utils"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+var printConnectionsHeader = []string{"Peer ID", "Multiaddr", "Opened", "Direction", "NumStreams"}
+
+func PrintPeerConnections(bus modules.Bus, routerType RouterType) error {
+	var (
+		connections     []libp2pNetwork.Conn
+		routerPlurality = ""
+	)
+
+	if routerType == AllRouterTypes {
+		routerPlurality = "s"
+	}
+
+	connections, err := getFilteredConnections(bus, routerType)
+	if err != nil {
+		return fmt.Errorf("getting connecions: %w", err)
+	}
+
+	if err := LogSelfAddress(bus); err != nil {
+		return fmt.Errorf("printing self address: %w", err)
+	}
+
+	// NB: Intentionally printing with `fmt` instead of the logger to match
+	// `utils.PrintPeerListTable` which does not use the logger due to
+	// incompatibilities with the tabwriter.
+	// (This doesn't seem to work as expected; i.e. not printing at all in tilt.)
+	if _, err := fmt.Fprintf(
+		os.Stdout,
+		"%s router peerstore%s:\n",
+		routerType,
+		routerPlurality,
+	); err != nil {
+		return fmt.Errorf("printing to stdout: %w", err)
+	}
+
+	if err := PrintConnectionsTable(connections); err != nil {
+		return fmt.Errorf("printing peer list: %w", err)
+	}
+	return nil
+}
+
+func PrintConnectionsTable(conns []libp2pNetwork.Conn) error {
```

Comment:
does `PrintConnectionsTable` need to be publically exposed given that we use `PrintPeerConnections` in the cli?